src/deploy/functions/validate.ts

```diff
@@ -49,3 +53,73 @@ export function functionIdsAreValid(functions: { id: string; platform: string }[
     throw new FirebaseError(msg);
   }
 }
+
+/**
+ * Validate secret environment variables setting, if any.
+ * A bad secret configuration can lead to a significant delay in function deploys.
+ *
+ * If validation fails for any secret config, throws a FirebaseError.
+ */
+export async function secretsAreValid(b: backend.Backend) {
+  const endpoints = backend
+    .allEndpoints(b)
+    .filter((e) => e.secretEnvironmentVariables && e.secretEnvironmentVariables.length > 0);
+  validatePlatformTargets(endpoints);
+  await validateSecretVersions(endpoints);
+}
+
+/**
+ * Ensures that all endpoints specifying secret environment variables target platform that supports the feature.
+ */
+function validatePlatformTargets(endpoints: backend.Endpoint[]) {
```

Comment:
We shouldn't disable a feature in v2 that exists in v1. In the future, people should be able to use the v2 params syntax in v1 (even if it looks wonky) and will be able to use the old string syntax instead of a param if they prefer to hard-code things and do it TheHardWay (TM)

Reply:
Using another thread to discuss plans for v2 support.